app/controllers/discourse_assign/assign_controller.rb

```diff
@@ -126,17 +126,27 @@ def group_members
         SQL
       end
 
+      group_assignment_count = Topic
+        .joins("JOIN assignments a ON a.topic_id = topics.id")
+        .where(<<~SQL, group_id: group.id)
+          a.assigned_to_id = :group_id AND a.assigned_to_type = 'Group'
+        SQL
+        .where("topics.deleted_at IS NULL")
+        .count
+
       assignment_count = Topic
-        .joins("JOIN assignments a ON a.topic_id = topics.id AND a.assigned_to_id IS NOT NULL ")
+        .joins("JOIN assignments a ON a.topic_id = topics.id")
         .where(<<~SQL, group_id: group.id)
-          (a.assigned_to_id IN (SELECT group_users.user_id FROM group_users WHERE group_id = :group_id) AND a.assigned_to_type = 'User')
-          OR
-          (a.assigned_to_id = :group_id AND a.assigned_to_type = 'Group')
+          a.assigned_to_id IN (SELECT group_users.user_id FROM group_users WHERE group_id = :group_id) AND a.assigned_to_type = 'User'
```

Comment:
Now that we don't have an `OR`, I think we can just do a LEFT JOIN on `group_users` which should be way more efficient.